src/pages/Fox/components/FoxWifHatClaim/FoxWifHatClaimConfirm.tsx

```diff
@@ -0,0 +1,256 @@
+  const accountNumber = useAppSelector(state =>
+    selectAccountNumberByAccountId(state, { accountId }),
+  )
+  const foxWifHatAsset = useAppSelector(state => selectAssetById(state, foxWifHatAssetId))
+  const getFoxWifHatClaimsQuery = useFoxWifHatMerkleTreeQuery()
+
+  const claimQuote = useMemo(() => {
+    const claim = getFoxWifHatClaimsQuery.data?.claims[fromAccountId(accountId).account]
+    if (!claim) return null
+
+    return claim
+  }, [getFoxWifHatClaimsQuery.data, accountId])
+
+  const amountCryptoPrecision = useMemo(() => {
+    if (!claimQuote) return
+    if (!foxWifHatAsset) return
+
+    return fromBaseUnit(claimQuote.amount, foxWifHatAsset.precision)
+  }, [claimQuote, foxWifHatAsset])
+
+  const callData = useMemo(() => {
+    if (!claimQuote) return '0x'
+    if (!foxWifHatAsset) return '0x'
+
+    return encodeFunctionData({
+      abi: FOX_WIF_HAT_MERKLE_DISTRIBUTOR_ABI,
+      functionName: 'claim',
+      args: [
+        claimQuote.index,
+        fromAccountId(accountId).account as `0x${string}`,
+        BigInt(claimQuote.amount),
+        claimQuote.proof,
+      ],
+    })
+  }, [claimQuote, accountId, foxWifHatAsset])
+
+  const {
+    mutateAsync: handleClaim,
+    isIdle: isClaimMutationIdle,
+    isPending: isClaimMutationPending,
+    isSuccess: isClaimMutationSuccess,
+  } = useMutation({
+    mutationFn: async () => {
+      if (!wallet || accountNumber === undefined) return
+
+      const adapter = assertGetEvmChainAdapter(fromAccountId(accountId).chainId)
+
+      const buildCustomTxInput = await createBuildCustomTxInput({
+        accountNumber,
+        from: fromAccountId(accountId).account,
+        adapter,
+        data: callData,
+        value: '0',
+        to: FOX_WIF_HAT_MERKLE_DISTRIBUTOR_CONTRACT,
+        wallet,
+      })
+
+      const txId = await buildAndBroadcast({
+        adapter,
+        buildCustomTxInput,
+        receiverAddress: CONTRACT_INTERACTION, // no receiver for this contract call
+      })
+
+      return txId
+    },
+    onSuccess: (txId: string | undefined) => {
+      if (!txId) return
+
+      setClaimTxid(txId)
+    },
+  })
+
+  const isGetClaimFeesEnabled = useMemo(() => Boolean(isClaimMutationIdle), [isClaimMutationIdle])
+
+  const claimFeesQueryInput = useMemo(
+    () => ({
+      to: FOX_WIF_HAT_MERKLE_DISTRIBUTOR_CONTRACT,
+      from: fromAccountId(accountId).account,
+      chainId: fromAccountId(accountId).chainId,
+      accountNumber,
+      data: callData,
+      value: '0',
+    }),
+    [accountNumber, accountId, callData],
+  )
+
+  const {
+    data: claimFees,
+    isLoading: isClaimFeesLoading,
+    isSuccess: isClaimFeesSuccess,
+  } = useEvmFees({
+    ...claimFeesQueryInput,
+    enabled: isGetClaimFeesEnabled,
+    staleTime: 30_000,
+    // Ensures fees are refetched at an interval, including when the app is in the background
+    refetchIntervalInBackground: true,
+    refetchInterval: isGetClaimFeesEnabled ? 15_000 : false,
+  })
```

Comment:
q: estimate call is failing for this guy using my main testing addy, and I'm getting 0 claim here, this doesn't seem correct?

<img width="1469" alt="image" src="https://github.com/user-attachments/assets/2959359c-f3ad-42f4-8b4a-8616cf362c7d" />

https://www.tdly.co/shared/simulation/73550b0f-f6b9-491a-8a51-70f108d2dc14

Reply:
Actually disregard me, was missing FWH asset locally